flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileReaderOperator.java

```diff
@@ -216,13 +216,13 @@ public void onNoMoreData(ContinuousFileReaderOperator<?> op) {
 		}
 	};
 
-	@VisibleForTesting
-	public ContinuousFileReaderOperator(FileInputFormat<OUT> format) {
-		this.format = checkNotNull(format);
-	}
+	public ContinuousFileReaderOperator(
+		FileInputFormat<OUT> format,
+		ProcessingTimeService processingTimeService,
+		MailboxExecutor mailboxExecutor) {
 
-	public ContinuousFileReaderOperator(FileInputFormat<OUT> format, MailboxExecutor mailboxExecutor) {
 		this.format = checkNotNull(format);
+		this.processingTimeService = checkNotNull(processingTimeService);
```

Comment:
This operator doesn't use `processingTimerService`. Why set this field?
The same goes for some other operators and their factories.
And if most operator factories don't use it then `AbstractStreamOperatorFactory` shouldn't have it.

Reply:
`AbstractStreamOperator` needs `ProcessingTimeService` ( see `StreamTaskStateInitializerImpl#streamOperatorState`, so almost all operators need it. For `ContinuousFileReaderOperator` , you can check traivis's failure log  to see this need:  https://travis-ci.com/flink-ci/flink/builds/148235635.